### ISSUE TITLE
Abstract store interface should be clarified as descriptive, not normative

### ISSUE THREAD

**Issue by jhamman** [zarr-developers/zarr-specs#307]
I would like to understand why the spec includes a [Abstract store interface](https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html#abstract-store-interface) section and ask if this section should be clarified as descriptive, not normative.

Why should the spec care what interface and implementation chooses for the Store abstraction? So long as the store does the right thing, surely the implementation is free to choose a different abstract interface?

**Comment by dcherian** [comment]
FWIW the wording is strong:

> For a store to be compatible with this specification, it must support a set of operations defined in the [Abstract store interface](https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html#id22) subsection.

**Comment by jbms** [comment]
Agreed that this is not intended to specify the actual API in any implementation.
